### PR TITLE
W-11341575: Spike on single operation execution

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/config/ConfigurationProviderObjectFactory.java [hunk 1]
```diff
@@ -52,18 +46,24 @@ class ConfigurationProviderObjectFactory extends AbstractExtensionObjectFactory<
...
 
+  private DslSyntaxResolver dslSyntaxResolver;
+
   ConfigurationProviderObjectFactory(ExtensionModel extensionModel,
                                      ConfigurationModel configurationModel,
                                      MuleContext muleContext) {
     super(muleContext);
     this.extensionModel = extensionModel;
     this.configurationModel = configurationModel;
+    dslSyntaxResolver = DslSyntaxResolver.getDefault(extensionModel,
```
Comment (elrodro83) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/config/ConfigurationProviderObjectFactory.java:65]: what is this needed here for?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/client/strategy/AbstractExtensionsClientProcessorsStrategy.java [hunk 2]
```diff
@@ -65,7 +67,7 @@ public AbstractExtensionsClientProcessorsStrategy(ExtensionManager extensionMana
     this.extensionManager = extensionManager;
     this.registry = registry;
     this.muleContext = muleContext;
-    this.policyManager = policyManager;
+    this.policyManager = policyManager != null ? policyManager : DEFAULT_POLICY_MANAGER;
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/client/strategy/AbstractExtensionsClientProcessorsStrategy.java:70]: Can we either make the policyManager param an Optional, or force the client code to provide the NoOp?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/ConfigurationCreationUtils.java [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+import org.mule.runtime.module.extension.internal.runtime.resolver.ConnectionProviderResolver;
+import org.mule.runtime.module.extension.internal.runtime.resolver.ConnectionProviderValueResolver;
+import org.mule.runtime.module.extension.internal.runtime.resolver.ResolverSet;
+import org.mule.runtime.module.extension.internal.util.ReflectionCache;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Utilities for creating {@link ConfigurationProvider} instances and their associated {@link ConnectionProviderResolver}\
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/ConfigurationCreationUtils.java:48]: \ at the end

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/ConfigurationCreationUtils.java [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+    return withContextClassLoader(extensionClassLoader, () -> {
+      ResolverSet resolverSet = getResolverSetFromStaticValues(
+                                                               configurationModel,
+                                                               parameters,
+                                                               muleContext,
+                                                               false,
+                                                               reflectionCache,
+                                                               expressionManager,
+                                                               parametersOwner,
+                                                               dslSyntaxResolver);
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/ConfigurationCreationUtils.java:96]: do we have a TODO to remove this code dependency on the dsl?

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/config/ConfigurationProviderObjectFactory.java (hunk 1 above)
Comment (marianogonzalez) [reply modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/config/ConfigurationProviderObjectFactory.java:65]: needed below

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/ConfigurationCreationUtils.java (hunk 4 above)
Comment (marianogonzalez) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/ConfigurationCreationUtils.java:96]: That's way too embedded into the logic and needed to support some edge cases. We're not looking at changing that at the moment

Comment (elrodro83) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/ConfigurationCreationUtils.java:96]: do we have a pending task to not lose track of it and do it sometime?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/client/strategy/AbstractExtensionsClientProcessorsStrategy.java (hunk 2 above)
Comment (marianogonzalez) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/client/strategy/AbstractExtensionsClientProcessorsStrategy.java:70]: I created a ticket to completly rewrite ExtensionClient and policy manager shouldn't exist here anymore. I'd rather just wait for that ticket than to tamper with this API

**Comment by marianogonzalez** [comment]
--validate

**Comment by marianogonzalez** [comment]
--validate

**Comment by marianogonzalez** [comment]
--validate

**Comment by marianogonzalez** [comment]
--validate

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**